mongo/integration/sessions_test.go

```diff
@@ -475,6 +475,43 @@ func TestSessions(t *testing.T) {
 		assert.True(mt, limitedSessionUse, limitedSessMsg, len(ops))
 
 	})
+
+	// Regression test for GODRIVER-2533. Note that this test assumes the race detector is enabled
+	// (GODRIVER-2072).
+	mt.Run("NumberSessionsInProgress data race", func(mt *mtest.T) {
+		// Start two goroutines under the same 100ms Context that continously run
+		// NumberSessionsInProgress and a basic collection operation
+		// (CountDocuments) simultaneously.
+		ctx, cancel := context.WithTimeout(context.Background(), 100*time.Millisecond)
+		defer cancel()
+
+		go func(ctx context.Context) {
+			for {
+				select {
+				case <-ctx.Done():
+					return
+				}
```

Comment:
This `select` statement seems like it will block until the `ctx.Done()` channel is closed and will never reach the call below. To get around that, either add a `default` case or check if the context is done in the `for` loop.

E.g.
```go
for ctx.Err() == nil {
	_ = mt.Client.NumberSessionsInProgress()
}
```

Reply:
I'm a little concerned about the busy waiting here as it occupies the CPU while waiting. Will this be an issue?

Reply:
TIL about the anti-pattern of [busy-waiting](https://en.wikipedia.org/wiki/Busy_waiting), thanks 🧑‍🔧 . I modified the test to only run each operation 5 times and used a channel to sync the goroutines' execution as opposed to a context. Note that the test fails now only about 75% of the time when `checkedOut` is not atomically accessed (as opposed to closer to 95% of the time with the previous design). I think that's fine, though.

Reply:
I think busy-wait loops in tests that are trying to exercise a specific non-deterministic behavior are probably OK, but I do think the "run N times" loops are a good way to achieve that also. I left some suggestions about possible ways to further improve the reliability of the tests in my subsequent review.